django/core/serializers/jsonl.py

```diff
@@ -39,19 +39,30 @@ def getvalue(self):
         return super(PythonSerializer, self).getvalue()
 
 
-def Deserializer(stream_or_string, **options):
+class Deserializer(PythonDeserializer):
     """Deserialize a stream or string of JSON data."""
-    if isinstance(stream_or_string, bytes):
-        stream_or_string = stream_or_string.decode()
-    if isinstance(stream_or_string, (bytes, str)):
-        stream_or_string = stream_or_string.split("\n")
-
-    for line in stream_or_string:
-        if not line.strip():
-            continue
+
+    def __init__(self, stream_or_string, **options):
+        if isinstance(stream_or_string, bytes):
+            stream_or_string = stream_or_string.decode()
+        if isinstance(stream_or_string, str):
+            stream_or_string = stream_or_string.splitlines()
+        super().__init__(Deserializer._get_lines(stream_or_string), **options)
+
+    def _handle_object(self, obj):
         try:
-            yield from PythonDeserializer([json.loads(line)], **options)
+            yield from super()._handle_object(obj)
         except (GeneratorExit, DeserializationError):
             raise
         except Exception as exc:
-            raise DeserializationError() from exc
+            raise DeserializationError(f"Error deserializing object: {exc}") from exc
```

Comment:
I can remove this `_handle_object` without any test failures

Reply:
Because of insufficient tests. For example, if you remove `_handle_object` in the json.py and run `python runtests.py fixtures_regress`, you'll get ` Animal has no field named 'extra_name'`. I will add corresponding tests for these scenartios for `jsonl.py` and `pyyaml.py`.